### DISCUSSION TITLE
Cookies are created with insecure identifier even though they have the secure flag

### DISCUSSION THREAD

**Discussion by justayar** [ory/hydra#2753] [Q&A] [answered]
Hello,

First of all, I would like to say that I am a complete newbie in Hydra.

We recently upgraded Hydra from 1.4 to 1.10.3. After the upgrade, although cookies appear secure in the browser panel
we have observed that cookies are created with the identifier **"insecure"** as follows.

<img width="1229" alt="Screenshot 2021-09-17 at 17 50 13" src="https://user-images.githubusercontent.com/7271001/133820767-4213f470-ed2b-44a0-8102-4740b61ae5c8.png">

We checked the changelogs and documentation and found the following description on the Csrf documentation page.

> ORY Hydra 1.7.0+ uses different names for cookies with and without the secure flag. In versions prior to this, you need to delete the domain cookies to get it working again.

We applied all the suggestions, but the situation was still the same. Meanwhile, we use **_Lax_** as SameSite mode in our configuration as below:

`SERVE_COOKIES_SAME_SITE_MODE: "Lax"`

When running hydra we pass the following parameters:
`
"serve all --dangerous-force-http"`

In conclusion, my question is, why are cookies created with the **"insecure"** identifier? Are we missing a point?

I will be grateful if you can help me.

Best regards!

**Comment by aeneasr** [comment, marked as answer]
You are running Ory Hydra with an insecure flag (`--dangerous-force-http` which enforces HTTP so all cookies will not have the secure flag!

**Reply by alfmos** [reply]
Hi @aeneasr , 
how are we supposed to manage secure csrf tokens behind a proxy?
Our hydra listens on localhost and it is behind nginx. Inside hydra.yml we put
```
  tls:
    allow_termination_from:
      - 127.0.0.1/32
```
and of course we set --dangerous-force-http, or hydra wouldn't start.
As a result csrf cookies are insecure (Samesite is lax), and every time hydra starts it tells us that using --dangerous-force-http is really dangerous, and we shouldn't.
What would be the correct configuration regarding cookies and http/https configuration? One that hydra finds well done?
thanks

**Reply by vinckr** [reply]
Hello @alfmos 
Maybe we should make a new discussion for this, but fwiw: 
Your Hydra is completely isolated from the outside internet? In that case --dangerous-force-http is not an issue IMO. Just the default for us is to assume that Hydra is exposed to the public internet, in which case you should be using https.

**Reply by alfmos** [reply]
Hi @vinckr,
yes, hydra sits behind an nginx proxy and is completely isolated from the Internet.
Even nginx sits behind a firewall that nats external addresses.
I am good with dangerous-force-http, what instead bothers me is that, to my understanding, this also forces CSRF cookies to be insecure. 
Actually, I would say that this configuration is way more secure, as nginx blocks most invalid requests, further protecting Hydra.
Is there a way to force cookies to be secure, as of course everything is in https, except the "last mile" from nginx to hydra?
Thanks
